tracsubtickets/web_ui.py

```diff
@@ -117,14 +117,14 @@ def get_children(self, parent_id, db=None):
     def validate_ticket(self, req, ticket):
         action = req.args.get('action')
         if action == 'resolve':
-            db = self.env.get_db_cnx()
-            cursor = db.cursor()
-            cursor.execute("SELECT parent, child FROM subtickets WHERE parent=%s",
-                           (ticket.id, ))
-
-            for parent, child in cursor:
-                if Ticket(self.env, child)['status'] != 'closed':
-                    yield None, _('Child ticket #%s has not been closed yet') % child
+            with self.env.db_transaction as db:
```

Comment:
Suggest changing queries per above coding style, however in absence of that, change `self.env.db_transaction` to `self.env.db_query` for any nested queries in which only `SELECT`s are done.